addons/stock_account/models/account_move_line.py

```diff
@@ -67,10 +67,19 @@ def _get_cogs_value(self):
             return self.product_id.standard_price
 
         # FIFO
-        moves = self._get_stock_moves()
+        moves = self._get_stock_moves().filtered(lambda m: m.state == 'done')
         if not moves:
             return self.product_id._run_fifo(self.quantity)
-        return moves._get_price_unit()
+
+        price_unit = moves._get_price_unit()
+        posted_cogs_value = sum(self.sale_line_ids.order_id.invoice_ids.line_ids.filtered(
+            lambda line: line.product_id == self.product_id and line.display_type == 'cogs'
+        ).mapped('debit'))
+        posted_cogs_qty = sum(self.sale_line_ids.order_id.invoice_ids.line_ids.filtered(
+            lambda line: line.product_id == self.product_id and line.display_type == 'cogs'
+        ).mapped('quantity')) / 2
```

Comment:
You could filter on the one having the valuation account and use the balance instead the debit (to avoid issue with negative value on return for example)